sympy/ntheory/factor_.py

```diff
@@ -17,7 +17,7 @@
 from sympy.core.power import integer_nthroot, Pow, integer_log
 from sympy.core.random import _randint
 from sympy.core.singleton import S
-from sympy.external.gmpy import SYMPY_INTS, gcd, lcm
+from sympy.external.gmpy import SYMPY_INTS, gcd, lcm, sqrt, sqrtrem
```

Comment:
Let's do here what was done in `qs.py`: import `sqrt` as `isqrt`. If the reader is fluent in gmpy they know this, but the more casual reader may not appreciate this as quickly. Then the few uses of `sqrt` throughout can be replaced with `isqrt`.